src/services/mongoose/wallet-invoices.ts

```diff
@@ -10,19 +14,17 @@ export const WalletInvoicesRepository = (): IWalletInvoicesRepository => {
     paid,
   }: WalletInvoice): Promise<WalletInvoice | RepositoryError> => {
     try {
-      await new InvoiceUser({
-        _id: paymentHash,
-        uid: walletId,
-        selfGenerated,
-        pubkey,
-        paid,
-      }).save()
+      const data = { uid: walletId, selfGenerated, pubkey, paid }
+      const doc = await InvoiceUser.findOneAndUpdate({ _id: paymentHash }, data, {
```

Comment:
We shouldn't reuse `persist` to update data. Better to introduce the pattern `persistNew` and `update` to all `Repositories`. This is more explicit. The calling code should always know which method is the right one to call.

Reply:
you suggested the current implementation previously https://github.com/GaloyMoney/galoy/pull/456#discussion_r692580344 

Please confirm that you want to change this